src/components/MiniWidgetContainer.vue

```diff
@@ -115,4 +117,15 @@ const refreshWidgetsHashs = (): void => {
 const showWidgetTrashArea = ref(false)
 
 const trashList = ref<MiniWidget[]>([])
+
+const handleDeleteWidget = (event: DraggableEvent): void => {
+  const widgetData = container.value.widgets.find((w) => w.hash === event.item.dataset.widgetHash)
+
+  // Remove VeryGenericIndicator from Logged variables list
+  if (widgetData?.component === 'VeryGenericIndicator') {
+    CurrentlyLoggedVariables.removeVariable(widgetData.options.displayName)
+  }
+
+  trashList.value = []
+}
```

Comment:
Thinking on a system-architecture point of view, we are adding a child-specific handler here on the parent.
What if we pass an event to the widget that is being deleted and ask him to deal with it? This way we keep the parent without child-specific code and also allow any mini-widget to decide what to do when it is removed.